docs/proposals/backend-support-appset.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Adding backend support for ApplicationSets RBAC CLI/Web requests
+authors:
+  - "@jgwest" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all intereste parties here.
+reviewers:
+  - "@alexmt"
+  - "@jannis"
+approvers:
+  - "@alexmt"
+  - "@jannis"
+
+creation-date: 2021-04-05
+last-updated: 2021-04-05
+---
+
+# Neat Enhancement Idea
+
+This is a proposal to add support for creating ApplicationSets via the Argo CD Web/CLI, by adding support to the ApplicationSet and API Server backend that respects Argo CD RBAC.
+
+## Summary
+
+Currently, users can only create ApplicationSets by applying configurations declaratively using `kubectl`. Introducing support for CLI/UI would improve the overall experience for managing large number of applications using ApplicationSet.
+
+## Motivation
+
+As ApplicationSet Controller is now part of ArgoCD installation, we would like to allow users to be able to create/update/delete application sets via CLI/UI.
+
+### Goals
+
+* **Expose endpoints in the API server to interact with ApplicationSets**
+
+  Users should be able to create/update/delete ApplicationSets using argocd CLI/UI.
+
+* **Changes to CLI**
+
+  Add a new argocd command option `appset` with `create`, `delete` and `update` sub-commands.
```

Comment:
should we add a `list` sub-command as well?

Reply:
Sure, looks like `app` has a list subcommand, so it couldn't hurt to add one for `appset` as well.